### DISCUSSION TITLE
How would I run something like: `wsl clang "C:\Users\Oliver\Desktop\hello.cpp"` (accessing a file in Windows in an app running in WSL2)? (my psuedo example returns no such file or dir)

### DISCUSSION THREAD

**Discussion by ItsCubeTime** [microsoft/WSL#6864] [Q&A]
Thanks <3

**Comment by Biswa96** [comment]
Try this `wsl clang $(wslpath "C:\Users\Oliver\Desktop\hello.cpp")`

**Reply by ItsCubeTime** [reply]
Thanks :) Works like a charm

Can WSL help with cross compiling from linux to Windows (like giving the compiler access to the correct libraries and such)?

**Reply by Biswa96** [reply]
Any example? FYI, one can do anything with a GNU/Linux environment.

**Reply by ItsCubeTime** [reply]
Not really sure I have an example. Im looking to compile C/C++ code into Windows executables in the Linux environment if possible (and viable).

**Reply by Biswa96** [reply]
Yes, absolutely. You just need `x86_64-w64-mingw32-` prefix compiler and toolchain. The package names may be different for each GNU/Linux distribution. If you are using ArchLinux or any pacman based distribution I can provide more help.

**Reply by ItsCubeTime** [reply]
> Yes, absolutely. You just need `x86_64-w64-mingw32-` prefix compiler and toolchain. The package names may be different for each GNU/Linux distribution. If you are using ArchLinux or any pacman based distribution I can provide more help.

Im using Ubuntu from the marketplace currently :)

The original idea was to try creating a symlink from Windows to WSL that will let me invoke my compiler (I prefer Clang, altough if thats not possible with mingw then thats okay) from the Windows cmd line like usual (as something like "clang <myfile> -Wall -Wextra") but with the actual compiler running in WSL2.

Any further knowledge would be greatly appreciated <3

**Reply by Biswa96** [reply]
The symlink idea is kind of sketchy. The `\\wsl$` thing has many issues here. Cross compiling in Windows for Windows is dumb. Native Win32 mingw compilers can be downloaded from anywhere.
